File: Day6/solution.js
const { SSL_OP_ALLOW_UNSAFE_LEGACY_RENEGOTIATION } = require('constants');
const fs = require('fs');
const data = fs.readFileSync('./input.txt', 'utf-8').split('\r\n\r\n')

// Part 1
const part1 = data.reduce((acc, group) => {
    let answers = new Set()
    let people = group.split('\r\n')

    for (person of people) {
        for (letter of person.split('')) {
            answers.add(letter)
        }
    }

    return acc + answers.size
}, 0)

console.log('Part 1: ', part1)

// Part 2
const part2 = data.reduce((acc, group) => {
    let people = group.split('\r\n')
    let groupSize = people.length;

    let counts = {}

    for (person of people) {
        let answers = person.split('');
        for (answer of answers) {
            counts[answer] = counts[answer] + 1 || 1
        }
    }
    
    let checkEntries = Object.entries(counts)
    let allAnswered = checkEntries.filter(e => {
        return e[1] === groupSize ? true : false
    })


    return acc + allAnswered.length

}, 0)

console.log('Part 2: ', part2)

File: Day6/solution.py
from collections import Counter

data = open('./input.txt').read().split('\n\n')

# Part 1
part1 = 0
for group in data:
    ans = group.replace('\n', '')
    unique_answers = set(list(ans))
    part1 += len(unique_answers)

print(f'Part 1: {part1}')

# Part 2
part2 = 0
for group in data:
    group_size = len(group.split('\n'))
    ans = group.replace('\n', '')
    counts = Counter(ans)
    for answer in counts:
        if counts[answer] == group_size:
            part2 += 1

print(f'Part 2: {part2}')
